aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
```

Comment:
One thing that can be added here are tuples. Probably something like `'(' type_list ')'` where `type_list: type [',' type_list]` or `type_list: type [',' type]*`.